### PR TITLE
extract: Add zstd support

### PR DESCRIPTION
**Pull request by cfergeau**
Testing[1] suggests that zstd compression might be better than xz, with
decompression being much faster than xz, so it's worth considering it
for our bundles.

There is no native support for zstd in golang, so this makes use of
https://pkg.go.dev/github.com/klauspost/compress/zstd for handling this
format. An alternative would be https://github.com/DataDog/zstd which
provides performance numbers, but which is not a pure go implementation,
which is why I picked the other one.

When compressed with `zstd --ultra -22 -T0`, the 4.6.9 bundle is 2.2G in
size, and uncompresses in about 1 minute. The xz-compressed bundle is
2.6G in size, and takes 3 minutes to uncompress.

An existing bundle can be converted to zstd for testing purposes with:
xzcat $bundle |zstd  -  -o crc_libvirt_xxx.crcbundle

This fixes https://github.com/code-ready/crc/issues/1340 for now.

## Testing

Behaviour should be unchanged, crc should now be able to use zstd-compressed
bundles when they are made available.

This can be tested by converting an existing bundle to zstd:
xzcat $bundle |zstd  -  -o crc_libvirt_xxx.crcbundle

and then cleaning up ~/.crc/cache, and running crc setup/crc start -b crc_libvirt_xxx.crcbundle

### CODE REVIEW

File: pkg/extract/extract.go
Comment (praveenkumar) [inline pkg/extract/extract.go:17]: Didn't go through the PR but as per document `Currently the package is heavily optimized for 64 bit processors and will be significantly slower on 32 bit processors.`  and some folks want to use crc on power-pc does it also affect performance there or they should be good using tarball of xz compression?

Comment (cfergeau) [reply pkg/extract/extract.go:17]: I expect most power machines to be 64 bit? Name of the arch is `ppc64le` after all. Not sure about s390, but I'd expect them to also be 64 bit.

Comment (gbraad) [reply pkg/extract/extract.go:17]: we do not support 32bits anyways, as Hyper-V is not available on those platforms (our image is 64bits), plus even PPC has been 64bits long before Intel+AMD started testing this.

**Review by gbraad — APPROVED** [review]

**Comment by guillaumerose** [comment]
Great! Let's wait few days for https://github.com/h2non/filetype/pull/98 maybe.

/lgtm

**Comment by cfergeau** [comment]
> Great! Let's wait few days for [h2non/filetype#98](https://github.com/h2non/filetype/pull/98) maybe.

The maintainer was very responsive, so yes we can wait for it (not merged yet as it needed a doc change which I have made now).

**Comment by cfergeau** [comment]
> > Great! Let's wait few days for [h2non/filetype#98](https://github.com/h2non/filetype/pull/98) maybe.
> 
> The maintainer was very responsive, so yes we can wait for it (not merged yet as it needed a doc change which I have now made).

It's merged now, I've amended the PR accordingly, and am testing the corresponding snc changes.

**Review by guillaumerose — APPROVED** [review]

**Review by gbraad — APPROVED** [review]

**Pull request merged**
